src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1502,6 +1502,52 @@ public static IEnumerable<string> ParseConditionalCompilationSymbols(string valu
             return defines.AsEnumerable();
         }
 
+        public static bool TryParseLanguageVersion(string value, out LanguageVersion version)
+        {
+            if (value == null)
+            {
+                version = LanguageVersion.Default;
+                return true;
+            }
+
+            switch (value.ToLowerInvariant())
+            {
+                case "iso-1":
+                    version = LanguageVersion.CSharp1;
+                    return true;
+
+                case "iso-2":
+                    version = LanguageVersion.CSharp2;
+                    return true;
+
+                case "default":
+                    version = LanguageVersion.Default;
+                    return true;
+
+                case "latest":
+                    version = LanguageVersion.Latest;
+                    return true;
+
+                default:
+                    if (float.TryParse(value, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out float versionNumber))
+                    {
+                        if ((int)versionNumber == versionNumber && 1 <= versionNumber && versionNumber <= 7)
+                        {
+                            version = (LanguageVersion)versionNumber;
+                            return true;
+                        }
+                        else if (versionNumber == 7.1f)
```

Comment:
I'll opt to using switches for point releases.
However, as a side point, the versioning work in this branch is not final as these changes will probably go to master (including some planning work for other features?) and will be merged here.

---
In reply to: [103756800](https://github.com/dotnet/roslyn/pull/17447#discussion_r103756800) [](ancestors = 103756800)